examples/classifier/CL2_apply_correlation_based_kmeans_clustering.py

```diff
@@ -0,0 +1,46 @@
+# Standard library imports
+import sys
+new_path = sys.path[0][:-len("/examples/timeseries")]
+sys.path.append(new_path)
+
+# Third party imports
+import seaborn as sns
+import matplotlib.pyplot as plt
+
+# Local application imports
+from scifin import timeseries as ts
+from scifin import statistics as st
+from scifin import montecarlo as mc
+from scifin import classifier as cl
+
+# Generate random walk processes
+mc1 = mc.generate_series(n=100, names_base="RW-", series_model=ts.random_walk,
+                         start_date="2000-01-01", end_date="2020-01-01", frequency='M',
```

Comment:
I haven't read through your codes in detail yet. But are you assuming the time indices of all your time series must be a date object? It seems a bit unnecessarily too restrictive

Reply:
Yes this is an assumption. I don't think it is too restrictive though. I can only think of some short time experiments where people may not care about the date, but in that case we can simply write functions that ease the index generation for the user, not rejecting the class definition. I think that for any kind of data recording of time series there should be an explicit specification of date and time. There may be cases you are thinking about that I have missed in my considerations though.

Reply:
Not sure I agree. There are also time series with unit of seconds/minutes, e.g. short selling in finance, heart rate or sensor monitor data. Also, sometimes there may not be an explicit datetime associated with the time series, but only time elapsed after certain epoch.

Given we are building algorithms for analysing (any) time series, I don't see why we should restrict ourselves to those time series associated with dates only

Reply:
We should discuss it further. Your examples are good examples indeed. Maybe a format with only time and no date would suffice. Since it is not directly related, let's keep that discussion going outside the PR.